include/deal.II/lac/petsc_ts.templates.h

```diff
@@ -442,7 +442,14 @@ namespace PETScWrappers
 
       const int lineno = __LINE__;
       const int err    = call_and_possibly_capture_ts_exception(
-        user->decide_for_coarsening_and_refinement,
+        (user->stop_for_mesh_adaptation ?
+              [&](const real_type    t,
+               const unsigned int step,
+               const VectorType  &y,
+               bool              &resize) {
+             resize = user->stop_for_mesh_adaptation(t, step, y);
+           } :
+              user->decide_for_coarsening_and_refinement),
```

Comment:
Not the most elegant way to indent this, but it's code that is going to go away after a couple of releases so I'm not bothered.